packages/verified-fetch/src/utils/parse-url-string.ts

```diff
@@ -31,6 +31,21 @@ export interface ParsedUrlStringResults {
 }
 
 const URL_REGEX = /^(?<protocol>ip[fn]s):\/\/(?<cidOrPeerIdOrDnsLink>[^/$?]+)\/?(?<path>[^$?]*)\??(?<queryString>.*)$/
+const PATH_REGEX = /^\/(?<protocol>ip[fn]s)\/(?<cidOrPeerIdOrDnsLink>[^/$?]+)\/?(?<path>[^$?]*)\??(?<queryString>.*)$/
```

Comment:
Why are the `$` necessary in the negated sets of the named capture groups, i.e., `[^/$?]` are we expecting $ in the URL?

Reply:
I'm not sure why they are there, it doesn't look like they are necessary and the tests pass without them, have removed.